### PR TITLE
Feat - migrate supabase UI. Move studio to turborepo

### PR DESCRIPTION
**Pull request by MildTomato**
## What kind of change does this PR introduce?

Bug fix, feature, docs update, ...

## What is the current behavior?

Please link any relevant issues here.

## What is the new behavior?

— Studio now part of turborepo, under the scope name `studio`
We have not moved the directory though to `apps/` as that would break existing PRs

— Moved `@supabase/ui` components into `/packages` under the package name `ui`

components are now imported like so:
```js
import { button } from 'ui'
```

turborepo local packages are installed in studio like this:
```js
// package.json
"dependencies": {
  // ..
  "ui": "*",
  "common": "*",
  // ..
},
```

## Additional context

Add any other context or screenshots.

- [ ] check self hosting
- [ ] smoke test UI in staging vercel

### CODE REVIEW

File: studio/.gitignore [hunk 1]
```diff
@@ -33,3 +33,4 @@ yarn-error.log*
 .env.production.local
 
 !.vercel
+.vercel
```
Comment (joshenlim) [inline studio/.gitignore:36]: i think this is conflicting with line 35?  do we want to ignore `.vercel`? or dont ignore `.vercel`?

**Review by joshenlim — APPROVED** [review]

**Comment by Hallidayo** [comment]
Hi @MildTomato - Is their anything different I need to do to run studio locally? I'm getting the following error going to localhost:8082:

Screenshot 2022-10-07 at 11 45 03

**Comment by J0** [comment]
Hey @Hallidayo the team recently moved to Turborepo -- you might want to try:


> ensure that supabase/studio has no package-lock
> rm package-lock from the root of supabase
> rm node_modules
> do an npm i
> npm run dev:studio

**Comment by Hallidayo** [comment]
Hi @J0 - Thankyou for the above, I've tried the steps but I'm still getting the same error, Here is is in my terminal:

Screenshot 2022-10-07 at 13 03 34

**Comment by J0** [comment]
Hmm if you can it might be a good idea to delete the existing repo, clone and do npm install again -- you may need to delete every node_modules folder in every app under /apps, and also in */studio

**Comment by Hallidayo** [comment]
@J0 - I'm all back up running now thank you so much.

I deleted out all the node_module folders and then re installed their must be have been a conflict somewhere.

**Pull request merged**